lib/services/lwm2mHandlers/updateRegistration.js

```diff
@@ -67,11 +67,20 @@ function updateRegistration(device, payload, callback) {
         });
     }
 
-    async.waterfall([
-        removePreviousSubscriptions,
-        apply(iotAgentLib.getDevice, device.name),
-        apply(commonLwm2m.observeActiveAttributes, payload)
-    ], callback);
+    /** If payload does not contain the list of objects/instances (usually this is only
+    * included if objects has changed since registration) */
+    if (!payload && payload === '') {
```

Comment:
be careful here with this condition: first of all an empty string is falsy per se so, at first glance, the condition should be:

```
if(!payload) 
```

however it would be good to do some kind of trimming before testing the condition 

```
if(!payload.trim()) 
```